web-registry/src/hooks/useQueryBalance.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect, useCallback } from 'react';
+
+import {
+  QueryBalanceResponse,
+  QueryClientImpl,
+} from '@regen-network/api/lib/generated/cosmos/bank/v1beta1/query';
+
+import { useLedger } from '../ledger';
+
+type FetchBalance = {
+  address: string | undefined;
+  denom: string | undefined;
+};
+
+export default function useQueryBalance(): Function {
```

Comment:
`Function` isn't the right type here. I believe it should be something like 

```ts
type FetchBalance = (args: Balance) => QueryBalanceResponse // string or whatever the return type of `await queryClient.Balance({ address, denom })` is
``` 
(note: i'm reusing your naming for the returned function)

which would change this line to
```suggestion
export function useQueryBalance(): FetchBalance {
```
or inline
```suggestion
export function useQueryBalance(): (args: Balance) => QueryBalanceResponse { 
```

and give type safety where it's invoked, ie here:
```ts
const fetchBalances = useQueryBalance()
fetchBalances({ address: 'someAddress', denoms: 'someDenom' })
```
you'd get type support with the instantiated `fetchBalances`. Typing as `Function` allows anything to be passed to it without TS complaining

Reply:
Thank you very much for your comments, now the function returned by the hook is better typed, thanks!